### PR TITLE
[Feat](Previsit) add multi session management

### PR DESCRIPTION
**Pull request by MeddahAbdellah**

### CODE REVIEW

File: src/ios/OpenTokPlugin.m [hunk 1]
```diff
@@ -403,7 +414,9 @@ - (void)subscribe:(CDVInvokedUrlCommand*)command{
     OTSubscriber* sub = [[OTSubscriber alloc] initWithStream:myStream delegate:self];
     sub.audioLevelDelegate = self;
     sub.networkStatsDelegate = self;
-    [_session subscribe:sub error:nil];
+    NSString* sessionId = [command.arguments objectAtIndex:10];
```
Comment (cadesalaberry) [inline src/ios/OpenTokPlugin.m:417]: move to the beginning of the function next to all the others `[command.arguments objectAtIndex`

File: src/ios/OpenTokPlugin.m [hunk 2]
```diff
@@ -450,7 +465,9 @@ - (void)signal:(CDVInvokedUrlCommand*)command{
     OTConnection* c = [connectionDictionary objectForKey: signalRecipientId];
 
     NSLog(@"iOS signaling to connection %@", c);
-    [_session signalWithType:signalType string:signalPayload connection:c error:nil];
+    NSString* sessionId = [command.arguments objectAtIndex:3];
```
Comment (cadesalaberry) [inline src/ios/OpenTokPlugin.m:468]: move to the beginning of the function next to all the others `[command.arguments objectAtIndex`

File: src/ios/OpenTokPlugin.m [hunk 3]
```diff
@@ -432,7 +445,9 @@ - (void)unsubscribe:(CDVInvokedUrlCommand*)command{
     NSString* sid = [command.arguments objectAtIndex:0];
     OTSubscriber * subscriber = [subscriberDictionary objectForKey:sid];
     @try {
-        [_session unsubscribe:subscriber error:nil];
+        NSString* sessionId = [command.arguments objectAtIndex:1];
```
Comment (cadesalaberry) [inline src/ios/OpenTokPlugin.m:448]: move to the beginning of the function next to all the others `[command.arguments objectAtIndex`

File: src/ios/OpenTokPlugin.m [hunk 4]
```diff
@@ -464,8 +481,10 @@ - (void)subscriberDidConnectToStream:(OTSubscriberKit*)sub{
     NSMutableDictionary* eventData = [[NSMutableDictionary alloc] init];
     NSString* streamId = sub.stream.streamId;
     [eventData setObject:streamId forKey:@"streamId"];
+    NSString* sessionId = sub.session.sessionId;
+    NSString* sessionEventName = [NSString stringWithFormat:@"sessionEvents:%@", sessionId];
     [self triggerJSEvent: @"subscriberEvents" withType: @"connected" withData: eventData];
-    [self triggerJSEvent: @"sessionEvents" withType: @"subscribedToStream" withData: eventData]; // Backwards compatibility
+    [self triggerJSEvent: sessionEventName withType: @"subscribedToStream" withData: eventData]; // Backwards compatibility
```
Comment (cadesalaberry) [inline src/ios/OpenTokPlugin.m:487]: Backwards compatibility with what ?

File: src/js/OTSession.coffee [hunk 5]
```diff
@@ -134,13 +134,14 @@ class TBSession
...
       TBUpdateObjects()
-    return Cordova.exec(TBSuccess, TBError, OTPlugin, "unsubscribe", [subscriber.streamId] )
+    return Cordova.exec(TBSuccess, TBError, OTPlugin, "unsubscribe", [subscriber.streamId, @sessionId] )
 
   constructor: (@apiKey, @sessionId) ->
     @apiKey = @apiKey.toString()
     @connections = {}
     @streams = {}
     @subscribers = {}
+    @capabilities = { publish: 1 }
```
Comment (cadesalaberry) [inline src/js/OTSession.coffee:144]: Mmm, should it always be 1 ?

File: src/js/OTSession.coffee [hunk 6]
```diff
@@ -186,7 +187,7 @@ class TBSession
     delete( @connections[ connection.connectionId] )
     return @
   sessionConnected: (event) =>
-    @connectionCompletedCallback?(null)
+    @connectionCompletedCallback?([null, { type: "sessionConnected" }])
```
Comment (cadesalaberry) [inline src/js/OTSession.coffee:190]: Shouldn't you return `new TBEvent("sessionConnected")` ?

File: src/js/OTSession.coffee (hunk 5 above)
Comment (MeddahAbdellah) [reply src/js/OTSession.coffee:144]: We are expecting 1 in the frontside
the web version returns capabilities, but the plugging doesn't so it's added arteficially

File: src/ios/OpenTokPlugin.m (hunk 4 above)
Comment (MeddahAbdellah) [reply src/ios/OpenTokPlugin.m:487]: I don't know, it's not my comment, it was here before I touched it

**Review by Orphidios — APPROVED** [review]

**Review by cadesalaberry — APPROVED** [review]

**Pull request merged**